File: app/src/main/java/com/ericho/coupleshare/mvp/data/LoginDataSource.java
package com.ericho.coupleshare.mvp.data;

/**
 * Created by steve_000 on 12/6/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp.data
 */

public interface LoginDataSource {

    interface LoginCallback{
        void onLoginSuccess();
        void onLoginFailure(Throwable t);
    }
    interface RegisterCallback{
        void onRegisterSuccess();
        void onRegisterFailure(Throwable t);
    }

    void login(String username,String password,LoginCallback callback);

    void register(String username,String password,RegisterCallback callback);

}

File: app/src/main/java/com/ericho/coupleshare/mvp/data/LoginDataSource.kt
package com.ericho.coupleshare.mvp.data

/**
 * Created by steve_000 on 6/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.mvp.data
 */
interface LoginDataSource {
    interface LoginCallback {
        fun onLoginSuccess()
        fun onLoginFailure(t: Throwable)
    }

    interface RegisterCallback {
        fun onRegisterSuccess()
        fun onRegisterFailure(t: Throwable)
    }

    fun login(username: String, password: String, callback: LoginCallback)

    fun register(username: String, password: String, callback: RegisterCallback)
}